plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/AppEngineLibraryContainerInitializer.java

```diff
@@ -100,6 +117,16 @@ public void initialize(IPath containerPath, IJavaProject project) throws CoreExc
     }
   }
 
+  private void validateJarPaths(LibraryClasspathContainer container) throws LibraryRepositoryServiceException {
+    IClasspathEntry[] classpathEntries = container.getClasspathEntries();
+    for (int i = 0; i < classpathEntries.length; i++) {
+      IClasspathEntry classpathEntry = classpathEntries[i];
+      if (!classpathEntry.getPath().toFile().exists()) {
+        classpathEntries[i] = repositoryService.getLibraryClasspathEntry(classpathEntry);
```

Comment:
`getLibraryClasspathEntry` seems a bit odd.  `refresh` or `rebuild` would seem to make more sense?

It seems a bit odd to have so much knowledge in this class.  Would it make sense to push this method into `LibraryClasspathContainer` itself:
```
if(container.revalidate(repositoryService)) {
   JavaCore.setClasspathContainer(…);
```